pangeo_forge_orchestrator/meta_types/bakery.py

```diff
@@ -0,0 +1,240 @@
+    "azure.centralus",
+    "azure.northcentralus",
+    "azure.westus",
+    "azure.southafricanorth",
+    "azure.centralindia",
+    "azure.eastasia",
+    "azure.japaneast",
+    "azure.jioindiawest",
+    "azure.koreacentral",
+    "azure.canadacentral",
+    "azure.francecentral",
+    "azure.germanywestcentral",
+    "azure.norwayeast",
+    "azure.switzerlandnorth",
+    "azure.uaenorth",
+    "azure.brazilsouth",
+    "azure.centralusstage",
+    "azure.eastusstage",
+    "azure.eastus2stage",
+    "azure.northcentralusstage",
+    "azure.southcentralusstage",
+    "azure.westusstage",
+    "azure.westus2stage",
+    "azure.asia",
+    "azure.asiapacific",
+    "azure.australia",
+    "azure.brazil",
+    "azure.canada",
+    "azure.europe",
+    "azure.global",
+    "azure.india",
+    "azure.japan",
+    "azure.uk",
+    "azure.unitedstates",
+    "azure.eastasiastage",
+    "azure.southeastasiastage",
+    "azure.centraluseuap",
+    "azure.eastus2euap",
+    "azure.westcentralus",
+    "azure.southafricawest",
+    "azure.australiacentral",
+    "azure.australiacentral2",
+    "azure.australiasoutheast",
+    "azure.japanwest",
+    "azure.jioindiacentral",
+    "azure.koreasouth",
+    "azure.southindia",
+    "azure.westindia",
+    "azure.canadaeast",
+    "azure.francesouth",
+    "azure.germanynorth",
+    "azure.norwaywest",
+    "azure.swedensouth",
+    "azure.switzerlandwest",
+    "azure.ukwest",
+    "azure.uaecentral",
+    "azure.brazilsoutheast",
+]
+s3_default_cache_types = Literal["bytes", "none"]
+
+# turn fsspec's list of known_implementations into object which pydantic can validate against
+KnownImplementations = Literal[tuple(list(known_implementations))]  # type: ignore
+
+FARGATE_CLUSTER = "aws.fargate"
+AKS_CLUSTER = "azure.aks"
+clusters = Literal[FARGATE_CLUSTER, AKS_CLUSTER]  # type: ignore
+
+PANGEO_FORGE_BAKERY_DATABASE = (
+    "https://raw.githubusercontent.com/pangeo-forge/bakery-database/main/bakeries.yaml"
+)
+# ensure that secrets are passed as env var names enclosed in curly braces, e.g. `"{MY_AWS_KEY}"`
+env_var_name = constr(regex=r"{(.*?)}")
+# accepts only strings of integer values 0-9
+run_identifier = constr(regex=r"^\d+$")
+# see https://github.com/pangeo-forge/roadmap/pull/34; the "@" delimiter is introduced here
+feedstock_name_with_version = constr(regex=r".*-feedstock@(0|[1-9]\d*)\.(0|[1-9]\d*)$")
+# TODO: possibly move recipe name validation into `pangeo_forge_orchestrator.meta_types.meta`
+# note that the string prior to the optional ":" delimiter must be a valid python identifier, but
+# the string after ":" is a dictionary key, and therefore does not have this requirement
+recipe_or_dictobj_identifier = constr(
+    regex=r"^([A-Za-z][A-Za-z0-9_]*)([:][A-Za-z][A-Za-z0-9_-]*)?$"
+)
+
+
+class StorageOptions(BaseModel):
+    key: Optional[env_var_name] = None  # type: ignore
+    secret: Optional[env_var_name] = None  # type: ignore
+    anon: Optional[bool] = None
```

Comment:
These options are unique to s3fs and won't work with gcsfs or adlfs. Need to generalize.